src/cluster.py

```diff
@@ -400,6 +381,21 @@ def switchover(self) -> None:
         if r.status_code != 200:
             raise SwitchoverFailedError(f"received {r.status_code}")
 
+    def failover(self) -> None:
+        """Trigger a failover."""
+        # Try to trigger the failover.
+        for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(3)):
+            with attempt:
+                r = requests.post(
+                    f"{self._patroni_url}/failover",
+                    json={"candidate": self.member_name},
+                    verify=self.verify,
+                )
```

Comment:
Can we please set `synchronous_node_count` to minority, instead of promoting unsync replica as primary from the charm code?
```
synchronous_node_count = self.app.planned_units() // 2
```

Reply:
Unfortunately, it doesn't seem like a minority count of sync standbys solves the instability in the `charm-integration` test. Since this PR has taken a while, I propose reverting to the default `synchronous_node_count` (1) and without automatic failover, break up `test_scale_down_and_up` into smaller tests and marking the parts that break as `unstable` until we have more clarity about how we expect things to behave. @marceloneppel @taurus-forever Would that be satisfactory for everyone?